### PR TITLE
docs(optimize): document SNI configuration options

### PR DESCRIPTION
**Pull request by buccarel**
## Description

Context:
https://github.com/camunda/camunda-optimize/issues/11941

With this PR, we are adding the documentation about the option to enable or disable the SNI check.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

### CODE REVIEW

File: optimize/self-managed/optimize-deployment/configuration/system-configuration.md [hunk 1]
```diff
@@ -125,6 +125,7 @@ Settings related to embedded Jetty container, which serves the Optimize applicat
 | container.status.connections.max | 10            | Maximum number of web socket connections accepted for status report.                                                                                                                                                                    |
 | container.accessUrl              | null          | Optional URL to access Optimize (used for links to Optimize in e.g. alert emails). If no value specified the container host and port are used instead.                                                                                  |
 | container.http2Enabled           | false         | Enable use of HTTP/2 for Optimize                                                                                                                                                                                                       |
+| container.enableSniCheck         | false         | Determines whether SNI checking should be enabled.                                                                                                                                                                                      |
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/configuration/system-configuration.md:128]: default is `true` 🙂

File: optimize_versioned_docs/version-3.10.0/self-managed/optimize-deployment/configuration/system-configuration.md [hunk 2]
```diff
@@ -125,6 +125,7 @@ Settings related to embedded Jetty container, which serves the Optimize applicat
 | container.status.connections.max | 10            | Maximum number of web socket connections accepted for status report.                                                                                                                                                                    |
 | container.accessUrl              | null          | Optional URL to access Optimize (used for links to Optimize in e.g. alert emails). If no value specified the container host and port are used instead.                                                                                  |
 | container.http2Enabled           | false         | Enable use of HTTP/2 for Optimize                                                                                                                                                                                                       |
+| container.enableSniCheck         | false         | Determines whether SNI checking should be enabled.                                                                                                                                                                                      |
```
Comment (RomanJRW) [inline optimize_versioned_docs/version-3.10.0/self-managed/optimize-deployment/configuration/system-configuration.md:128]: same as above

File: optimize_versioned_docs/version-3.11.0/self-managed/optimize-deployment/configuration/system-configuration.md [hunk 3]
```diff
@@ -125,6 +125,7 @@ Settings related to embedded Jetty container, which serves the Optimize applicat
 | container.status.connections.max | 10            | Maximum number of web socket connections accepted for status report.                                                                                                                                                                    |
 | container.accessUrl              | null          | Optional URL to access Optimize (used for links to Optimize in e.g. alert emails). If no value specified the container host and port are used instead.                                                                                  |
 | container.http2Enabled           | false         | Enable use of HTTP/2 for Optimize                                                                                                                                                                                                       |
+| container.enableSniCheck         | false         | Determines whether SNI checking should be enabled.                                                                                                                                                                                      |
```
Comment (RomanJRW) [inline optimize_versioned_docs/version-3.11.0/self-managed/optimize-deployment/configuration/system-configuration.md:128]: same as above

File: optimize_versioned_docs/version-3.12.0/self-managed/optimize-deployment/configuration/system-configuration.md [hunk 4]
```diff
@@ -125,6 +125,7 @@ Settings related to embedded Jetty container, which serves the Optimize applicat
 | container.status.connections.max | 10            | Maximum number of web socket connections accepted for status report.                                                                                                                                                                    |
 | container.accessUrl              | null          | Optional URL to access Optimize (used for links to Optimize in e.g. alert emails). If no value specified the container host and port are used instead.                                                                                  |
 | container.http2Enabled           | false         | Enable use of HTTP/2 for Optimize                                                                                                                                                                                                       |
+| container.enableSniCheck         | false         | Determines whether SNI checking should be enabled.                                                                                                                                                                                      |
```
Comment (RomanJRW) [inline optimize_versioned_docs/version-3.12.0/self-managed/optimize-deployment/configuration/system-configuration.md:128]: same as above

File: optimize_versioned_docs/version-3.13.0/self-managed/optimize-deployment/configuration/system-configuration.md [hunk 5]
```diff
@@ -125,6 +125,7 @@ Settings related to embedded Jetty container, which serves the Optimize applicat
 | container.status.connections.max | 10            | Maximum number of web socket connections accepted for status report.                                                                                                                                                                    |
 | container.accessUrl              | null          | Optional URL to access Optimize (used for links to Optimize in e.g. alert emails). If no value specified the container host and port are used instead.                                                                                  |
 | container.http2Enabled           | false         | Enable use of HTTP/2 for Optimize                                                                                                                                                                                                       |
+| container.enableSniCheck         | false         | Determines whether SNI checking should be enabled.                                                                                                                                                                                      |
```
Comment (RomanJRW) [inline optimize_versioned_docs/version-3.13.0/self-managed/optimize-deployment/configuration/system-configuration.md:128]: same as above

**Review by RomanJRW — CHANGES_REQUESTED** [review]
Thanks for doing this, just a change in the default values please 🙂

**Comment by akeller** [comment]
@buccarel since we have an alpha release tomorrow, we'll do doing at least one docs release this week (later today). If you get this cleaned up, we can include it!

**Comment by RomanJRW** [comment]
@buccarel has FTO for the next few days. Given that this is not new functionality, I don't see the need to find someone on the team to take over in his absence and it can wait until his return

**Review by RomanJRW — APPROVED** [review]

**Pull request merged**